torch/distributed/benchmarks/benchmark_ddp_rpc.py

```diff
@@ -0,0 +1,350 @@
+import argparse
+import io
+import os
+import random
+import shlex
+import subprocess
+import time
+
+import numpy as np
+import torch
+import torch.distributed as dist
+import torch.distributed.autograd as dist_autograd
+import torch.distributed.rpc as rpc
+import torch.multiprocessing as mp
+import torch.optim as optim
+from torch.distributed.optim import DistributedOptimizer
+from torch.distributed.rpc import RRef, TensorPipeRpcBackendOptions
+from torch.distributed.rpc.backend_registry import BackendType
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+
+# Config
+NUM_TRAINERS = 8
+NUM_PS = 8
+
+NUM_EMBEDDINGS = 100
+EMBEDDING_DIM = 16
+
+WARMUP_CYCLES = 5
+
+
+class HybridModel(torch.nn.Module):
+    r"""
+   The model consists of a sparse part and a dense part. The dense part is an
+   nn.Linear module that is replicated across all trainers using
+   DistributedDataParallel. The sparse part has nn.EmbeddingBags stored on multiple
+   parameter servers.
+
+   The model holds a Remote Reference to the embedding tables on the parameter
+   servers.
+   """
+
+    def __init__(self, emb_rref_list, device):
+        super(HybridModel, self).__init__()
+        self.emb_rref_list = emb_rref_list
+        self.fc = DDP(
+            torch.nn.Linear(NUM_PS * EMBEDDING_DIM, 8).cuda(device), device_ids=[device]
```

Comment:
We should also have maybe 3-5 linear layers stacked one after the other to simulate a MLP.